### PR TITLE
AB#81874 unbind allocated memory from set timeout

### PR DESCRIPTION
**Pull request by RenzoPrats**
# Description

Feat: Added destroy to setTimeout  function that returns a listener in order to prevent memory leaks.

## Useful links

- Please insert link to ticket

## Type of change

- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

It has been tested verifying if the changed files continue working correctly.

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: apps/back-office/src/app/dashboard/pages/reference-data/reference-data.component.ts [hunk 1]
```diff
@@ -819,6 +834,15 @@ export class ReferenceDataComponent
...
+    if (this.addChipListTimeoutListener) {
+      clearTimeout(this.addChipListTimeoutListener);
+    }
+    if (this.initEditorTimeoutListener) {
+      clearTimeout(this.initEditorTimeoutListener);
+    }
+    if (this.formTimeoutListener) {
+      clearTimeout(this.formTimeoutListener);
+    }
     super.ngOnDestroy();
```
Comment (unai-reliefapp) [inline apps/back-office/src/app/dashboard/pages/reference-data/reference-data.component.ts]: Hi @RenzoPrats , I see that you set the timeout unbinding in a lot of places before all the extended classes ngOnDestroy call, is there a reason for it?
If not I'll update this PR to move those extended classes calls to be first in the ngOnDestroy

KR, Unai

Comment (RenzoPrats) [reply apps/back-office/src/app/dashboard/pages/reference-data/reference-data.component.ts]: Hi @unai-reliefapp, there's no specific reason I just tried to follow the examples we have in others place of the code, where we always clear the timeout before call it and then also clear in the ngOnDestroy.

Comment (unai-reliefapp) [reply apps/back-office/src/app/dashboard/pages/reference-data/reference-data.component.ts]: No problem, anyway I'll use this PR to improve the timeout set of the snackbar, I think that most of the problem is there, there are a lot displayed in the application while using it and currently there are not correctly unbind

Thanks Renzo!

**Comment by unai-reliefapp** [comment]
@RenzoPrats  Added some missing timeout listeners and unsubscribes
Fix all the snackbars memory allocation problem 👍 
Snackbars still working as expected

Manual dismiss:
https://github.com/ReliefApplications/ems-frontend/assets/123092672/0b5a4fd0-7697-4440-bea7-391623477780

Automatic dismiss:
https://github.com/ReliefApplications/ems-frontend/assets/123092672/39534279-a986-48f0-b7b3-c12bbe84ffed

**Review by unai-reliefapp — APPROVED** [review]

**Pull request merged**